neighsyncd/neighsync.cpp

```diff
@@ -16,12 +16,33 @@
 using namespace std;
 using namespace swss;
 
-NeighSync::NeighSync(RedisPipeline *pipelineAppDB) :
+NeighSync::NeighSync(RedisPipeline *pipelineAppDB, DBConnector *stateDb) :
     m_neighTable(pipelineAppDB, APP_NEIGH_TABLE_NAME),
+    m_stateNeighRestoreTable(stateDb, STATE_NEIGH_RESTORE_TABLE_NAME),
     m_AppRestartAssist(pipelineAppDB, "neighsyncd", "swss", &m_neighTable, DEFAULT_NEIGHSYNC_WARMSTART_TIMER)
 {
 }
 
+// Check if neighbor table is restored in kernel
+bool NeighSync::isNeighRestoreDone()
+{
+    vector<FieldValueTuple> temp;
+
+    if (m_stateNeighRestoreTable.get("Flags", temp))
+    {
+        for (auto it : temp)
+        {
+            if (it.first == "restored" && it.second == "true")
+            {
```

Comment:
it is optional, but Table::hget() may be used to simplify the code.

Reply:
good suggestion, changed it.